luigi/worker.py

```diff
@@ -170,14 +170,25 @@ def run(self):
     def _recursive_terminate(self):
         import psutil
 
-        parent = psutil.Process(self.pid)
-        for child in parent.children(recursive=True):
-            child.kill()
+        try:
+            parent = psutil.Process(self.pid)
+            children = parent.children(recursive=True)
+
+            # terminate parent. Give it a chance to clean up
+            super(TaskProcess, self).terminate()
+            parent.wait()
 
-        parent.kill()
+            # terminate children
+            for child in children:
+                try:
+                    child.terminate()
+                except psutil.NoSuchProcess:
+                    continue
+        except psutil.NoSuchProcess:
+            return
```

Comment:
This outer try/except catches the case where assigning parent or children fails (and, potentially, parent.wait() but i'm not sure). This most often happens if the parent finishes before we enter try block. In that case, there's nothing to do, so we exit the method